dotCMS/src/main/java/com/dotcms/publisher/receiver/BundlePublisher.java

```diff
@@ -193,65 +193,58 @@ public PublisherConfig process ( final PublishStatus status ) throws DotPublishi
-
-                //Update audit
-                try {
-                    detail.setStatus(PublishAuditStatus.Status.FAILED_TO_PUBLISH.getCode());
-                    detail.setInfo("Failed to publish because an error occurred: " + e.getMessage());
-                    detail.setStackTrace(ExceptionUtils.getStackTrace(e));
-                    String endPointId = (String) currentStatusHistory.getEndpointsMap().keySet().toArray()[0];
-                    currentStatusHistory.addOrUpdateEndpoint(endPointId, endPointId, detail);
-                    currentStatusHistory.setPublishEnd(new Date());
-                    currentStatusHistory.setAssets(assetsDetails);
-
-                    auditAPI.updatePublishAuditStatus(bundleFolder, PublishAuditStatus.Status.FAILED_TO_PUBLISH,
-                            currentStatusHistory);
-                } catch (DotPublisherException e1) {
-                    throw new DotPublishingException("Cannot update audit: ", e);
-                }
-                throw new DotPublishingException("Error Publishing: " + e, e);
+                HibernateUtil.rollbackTransaction();
+            } catch (DotHibernateException e1) {
+                Logger.error(PublisherAPIImpl.class, e.getMessage(), e1);
             }
+            Logger.error(PublisherAPIImpl.class, "Error publishing bundle with ID '" + bundleName + "': " + e.getMessage(), e);
 
+            //Update audit
             try {
-                //Update audit
-                detail.setStatus(PublishAuditStatus.Status.SUCCESS.getCode());
-                detail.setInfo("Everything ok");
+                detail.setStatus(PublishAuditStatus.Status.FAILED_TO_PUBLISH.getCode());
+                detail.setInfo("Failed to publish because an error occurred: " + e.getMessage());
+                detail.setStackTrace(ExceptionUtils.getStackTrace(e));
                 String endPointId = (String) currentStatusHistory.getEndpointsMap().keySet().toArray()[0];
                 currentStatusHistory.addOrUpdateEndpoint(endPointId, endPointId, detail);
                 currentStatusHistory.setPublishEnd(new Date());
                 currentStatusHistory.setAssets(assetsDetails);
-                auditAPI.updatePublishAuditStatus(bundleFolder, PublishAuditStatus.Status.SUCCESS, currentStatusHistory);
-            } catch (Exception e) {
-                Logger.error(BundlePublisher.class, "Unable to update audit table : " + e.getMessage(), e);
+
+                auditAPI.updatePublishAuditStatus(bundleFolder, PublishAuditStatus.Status.FAILED_TO_PUBLISH,
+                        currentStatusHistory);
+            } catch (DotPublisherException e1) {
+                throw new DotPublishingException("Cannot update audit of bundle with ID '" + bundleName + "': ", e);
             }
+            throw new DotPublishingException("Error publishing bundle with ID '" + bundleName + "': " + e, e);
         } finally {
             DbConnectionFactory.closeSilently();
```

Comment:
Keep in mind you only can close a connection that you open!

Reply:
I just added more data to the error message, did not change the code.